lxd/migrate.go

```diff
@@ -533,26 +533,53 @@ type MigrationSinkArgs struct {
 	Push      bool
 }
 
-func NewMigrationSink(args *MigrationSinkArgs) (*migrationSink, error) {
+func NewMigrationSink(args *MigrationSinkArgs, push bool) (*migrationSink, error) {
 	sink := migrationSink{
 		src:    migrationFields{container: args.Container},
 		url:    args.Url,
 		dialer: args.Dialer,
+		push:   args.Push,
 	}
 
-	var ok bool
+	if push {
+		sink.allConnected = make(chan bool, 1)
+	}
+
+	var (
+		ok  bool
+		err error
+	)
```

Comment:
Just nitpicking but we've never used this style of variable definition in our code so far and I'm not sure why this change is needed at all since "err" is never accessed from outside a block below (so doesn't need to be defined outside).